### PR TITLE
Update registry to 2.8.0-beta.1

### PR DESCRIPTION
**Pull request by crazy-max**
https://github.com/distribution/distribution/releases/tag/v2.8.0-beta.1

cc @milosgajdos @thaJeztah 

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Review by milosgajdos — APPROVED** [review]

File: amd64/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,13 @@
-# Build a minimal distribution container
+FROM alpine:3.14 AS download
+RUN apk add --no-cache tar wget
+WORKDIR /out
+RUN wget -qO- https://github.com/distribution/distribution/releases/download/v2.8.0-beta.1/registry_2.8.0-beta.1_linux_amd64.tar.gz | tar -zxvf - registry
```
Comment (thaJeztah) [inline amd64/Dockerfile:4]: Perhaps for the "ga" release, we should use a `ARG` for the version

**Review by thaJeztah — COMMENTED** [review]
I see permissions changed on one file that an issue? (Haven't checked the other / new `.sh` files) 

Screenshot 2022-01-21 at 18 29 50

**Comment by tianon** [comment]
Nice, congrats on getting official binaries on https://github.com/distribution/distribution/releases/tag/v2.8.0-beta.1!  That's huge. :partying_face:

My confusion with this PR is that it's still one `Dockerfile` per architecture, the use of a multi-stage build seems unnecessary, and we could/should be verifying the released checksums too. :sweat_smile:

Here's an alternative proposal:

```dockerfile
FROM alpine:3.15

RUN apk add --no-cache ca-certificates

RUN set -eux; \
# https://github.com/distribution/distribution/releases
	version='2.8.0-beta.1'; \
	apkArch="$(apk --print-arch)"; \
	case "$apkArch" in \
		x86_64)  arch='amd64';   sha256='d69d68406a30466070d8e47c21fafaadae5c048239df41b50b28a79fa12945ab' ;; \
		aarch64) arch='arm64';   sha256='2a080694ae2528a628245cf5177d4d2e55d430c5dd76cbbf61e5aee08e75abf6' ;; \
		armhf)   arch='armv6';   sha256='b5092556fd196f59c7d2a8d4e4460c193588d9f7f39d825d17f299d7fc856ca1' ;; \
		armv7)   arch='armv7';   sha256='f479d7d42a4c6086ee9a51f605386fcfb953198bf9ab48c515ffdde33ad46e5d' ;; \
		ppc64le) arch='ppc64le'; sha256='dc0444e672511b4f8dc19fcec1624a33fdaf49a72ff3c1a69bae9bd3399cd074' ;; \
		s390x)   arch='s390x';   sha256='4b814d1cb60ee7881e59c1ee52635755c2279196861892cface92e58aa6ac749' ;; \
		*) echo >&2 "error: unsupported architecture: $apkArch"; exit 1 ;; \
	esac; \
	wget -O registry.tar.gz "https://github.com/distribution/distribution/releases/download/v${version}/registry_${version}_linux_${arch}.tar.gz"; \
	echo "$sha256 *registry.tar.gz" | sha256sum -c -; \
	tar --extract --verbose --file registry.tar.gz --directory /bin/ registry; \
	rm registry.tar.gz; \
	registry --version

COPY ./config-example.yml /etc/docker/registry/config.yml

VOLUME ["/var/lib/registry"]
EXPOSE 5000

COPY docker-entrypoint.sh /entrypoint.sh
ENTRYPOINT ["/entrypoint.sh"]

CMD ["/etc/docker/registry/config.yml"]
```

**Comment by thaJeztah** [comment]
@tianon is BuildKit used in the pipeline (i.e., could the `TARGETARCH` be used? (that could keep it even simpler for the downloads)

**Comment by crazy-max** [comment]
@tianon Thanks yes indeed we should merge that for GA!

**Comment by crazy-max** [comment]
> @tianon is BuildKit used in the pipeline (i.e., could the `TARGETARCH` be used? (that could keep it even simpler for the downloads)

Yes if BuildKit is used we could simply do:

```dockerfile
ARG REGISTRY_VERSION=latest
FROM distribution/distribution:${REGISTRY_VERSION} AS registry

FROM alpine:3.14

RUN set -ex && apk add --no-cache ca-certificates

COPY --from=registry /bin/registry /bin/registry
COPY ./config-example.yml /etc/docker/registry/config.yml

VOLUME ["/var/lib/registry"]
EXPOSE 5000

COPY docker-entrypoint.sh /entrypoint.sh
ENTRYPOINT ["/entrypoint.sh"]

CMD ["/etc/docker/registry/config.yml"]
```

**Pull request merged**